docs/adr/0001-user-roles.rst

```diff
@@ -0,0 +1,51 @@
+ADR 0001 - User roles
+=====================
+
+Status
+------
+
+Proposed
+
+
+Context
+-------
+
+Depending on what we expect them to do on Marcha, we need to define some user roles.
+
+First we have the people managing a Marcha instance.
+
+Then we have people linking their own website (this website is a "consumer site") to a Marcha instance, to host videos.
+
+Next we have the video authors.
+
+And finally we have the users coming to watch videos.
+
+
+Decision
+--------
+
+For the first group of people, the ``is_superuser`` flag on the Django User model is enough.
+
+For the second group of people, we first need a Django model to host the link between the Marcha instance and the consumer. We name this model ``ConsumerSite``.
+
+This ``ConsumerSite`` model can have many admins (to avoid having no admin if the only one existing is not available anymore). So there is a ``ManyToMany`` link between this model and the ``User`` model. Having this link make the user an admin on the consumer site.
+
+It's not in the scope of this document to decide how the entries in the ``ConsumerSite`` model and the admins (and the links between both) are created.
+
+For the third group of people, the authors, we don't need them to create an account, but they still have one: when coming from the consumer site, to upload a video, the link/form to do so contains a token unique for this author (the same every time for the same user, or if not the same, it will still allow to identify the same user) that will create the account on Marcha the first time, and authenticate it every time. Like a SSO.
+
+It has to be the same way to manage the videos: comming from the consumer site with this token. Maybe later if needed we can add a way for the author to get direct access to its account but it's not in the current scope.
+
+And finally for the last group of people, the users watching videos. We don't need them to be real users in the database. Each time a user does an action to view a video on the consumer site, a unique token, with a limited life span is generated, and is then allowed to see the given video.
+
+This is not the scope of this document to adress token generations.
+
+
+Consequences
+------------
+
+Simple for users: we avoid asking then to create their accounts on the Marcha instance (except for admins).
```

Comment:
It's what I called admins in the rest of the documents but it will be more clear after resolving the discussion about theses "organizations managers" in the other thread